nova/compute/manager.py

```diff
@@ -2209,7 +2209,7 @@ def _do_build_and_run_instance(self, context, instance, image,
                     image, filter_properties, admin_password,
                     injected_files, requested_networks, security_groups,
                     block_device_mapping, request_spec=request_spec,
-                    host_lists=[host_list])
+                    host_lists=[host_list], last_seen_error_message=e.kwargs['reason'])
```

Comment:
It may be safer to pull this in a way where it can't raise a KeyError. Is this the only way to get `reason`? I haven't seen this syntax before on an exception.

```python
e.kwargs.get('reason')
```

Reply:
This is the RescheduledException defined in nova. (https://github.com/ChameleonCloud/nova/blob/chameleoncloud/train/nova/exception.py#L1658-L1660) it has "instance_uuid" and "reason" parameters. They use this syntax to get the reschedule reason several lines before as well. (https://github.com/ChameleonCloud/nova/blob/chameleoncloud/train/nova/compute/manager.py#L2147) .

Reply:
Ah, OK 👍